core/services/job/orm.go

```diff
@@ -78,6 +78,8 @@ type ORM interface {
 
 	DataSource() sqlutil.DataSource
 	WithDataSource(source sqlutil.DataSource) ORM
+
+	FindJobIDByWorkflow(ctx context.Context, spec WorkflowSpec) (int32, error)
```

Comment:
nit: a fully-descriptive name would be FindJobByWorkflowNameAndOwner but it's annoyingly long.